### PR TITLE
New data source: aws_wafv2_web_acl

### LINKED ISSUE #11181: Feature Request: WAFv2 Web ACL Data Source
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In November 2019, AWS released a new version of the WAF API, [WAFv2](https://docs.aws.amazon.com/waf/latest/APIReference/Welcome.html), which offers improved functionality over the previous WAF API ("WAF Classic") such as [Managed Rules](https://github.com/terraform-providers/terraform-provider-aws/issues/11046) and WAF Capacity Units. This new API requires separate Terraform resource implementations from the previous resource implementations.

### New Data Source

* aws_wafv2_web_acl

### Potential Terraform Configuration

```hcl
data "aws_wafv2_web_acl" "example" {
  name = ""
  scope = ""
}
```

### References

- Requires: https://github.com/terraform-providers/terraform-provider-aws/pull/11172
- API Reference: https://docs.aws.amazon.com/waf/latest/APIReference/API_Operations_AWS_WAFV2.html
- SDK Reference: https://docs.aws.amazon.com/sdk-for-go/api/service/wafv2/

### PR DESCRIPTION
**Pull request by pvanbuijtene**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #11181
Relates #11046
Depends on #12688 because of duplicated code for tests.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
**New Data Source:** `aws_wafv2_web_acl` ()
```

Output from acceptance testing:

```
make build; make testacc TEST=./aws TESTARGS='-run=TestAccDataSourceAwsWafv2WebACL'
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccDataSourceAwsWafv2WebACL -timeout 120m
=== RUN   TestAccDataSourceAwsWafv2WebACL_Basic
=== PAUSE TestAccDataSourceAwsWafv2WebACL_Basic
=== CONT  TestAccDataSourceAwsWafv2WebACL_Basic
--- PASS: TestAccDataSourceAwsWafv2WebACL_Basic (38.61s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       39.052s
```

### CODE REVIEW

**Comment by anGie44** [comment]
dropping a note here to rebase and isolate the data source changes when you get a chance @pvanbuijtene :)

File: aws/data_source_aws_wafv2_web_acl_test.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccDataSourceAwsWafv2WebACL_NonExistent(name),
+				ExpectError: regexp.MustCompile(`WAFV2 WebACL not found`),
+			},
+			{
+				Config: testAccDataSourceAwsWafv2WebACL_Name(name),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair(datasourceName, "arn", resourceName, "arn"),
```
Comment (anGie44) [inline aws/data_source_aws_wafv2_web_acl_test.go:28]: we should also use the ARN testing check function in this step

File: aws/provider.go [hunk 2]
```diff
@@ -881,6 +882,7 @@ func Provider() terraform.ResourceProvider {
 			"aws_wafregional_xss_match_set":                           resourceAwsWafRegionalXssMatchSet(),
 			"aws_wafregional_web_acl":                                 resourceAwsWafRegionalWebAcl(),
 			"aws_wafregional_web_acl_association":                     resourceAwsWafRegionalWebAclAssociation(),
+			"aws_wafv2_web_acl":                                       resourceAwsWafv2WebACL(),
```
Comment (anGie44) [inline aws/provider.go:885]: note to remove as it will be in the resource PR

**Comment by anGie44** [comment]
hi @pvanbuijtene, dropping a note here to please rebase :)

File: website/docs/d/wafv2_web_acl.html.markdown [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+* `name` - (Required) The name of the WAFv2 Web ACL.
+* `scope` - (Required) Specifies whether this is for an AWS CloudFront distribution or for a regional application. Valid values are `CLOUDFRONT` or `REGIONAL`. To work with CloudFront, you must also specify the region `us-east-1` (N. Virginia) on the AWS provider.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `arn` - The Amazon Resource Name (ARN) of the entity.
+* `description` - The description of the set that helps with identification.
```
Comment (anGie44) [inline website/docs/d/wafv2_web_acl.html.markdown:34]:
```suggestion
* `description` - The description of the WebACL that helps with identification.
```

File: website/docs/d/wafv2_web_acl.html.markdown [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+* `id` - The unique identifier for the set.
```
Comment (anGie44) [inline website/docs/d/wafv2_web_acl.html.markdown:35]:
```suggestion
* `id` - The unique identifier for the WebACL.
```

**Comment by pvanbuijtene** [comment]
@anGie44 the changes are applied.

**Pull request merged**